pkg/venafi/cloud/connector.go

```diff
@@ -1274,34 +1267,55 @@ type managedCertificate struct {
 }
 
 func (c *Connector) getCertificate(certificateId string) (*managedCertificate, error) {
-	var err error
 	url := c.getURL(urlResourceCertificateByID)
 	url = fmt.Sprintf(url, certificateId)
-	statusCode, _, body, err := c.request("GET", url, nil)
-	if err != nil {
-		return nil, err
-	}
 
-	switch statusCode {
-	case http.StatusOK:
-		var res = &managedCertificate{}
-		err = json.Unmarshal(body, res)
+	// TODO: Remove following retry logic once VC-31590 is fixed
+	// retry logic involves the loop to constantly, during 1 minute, to retry
+	// to get certificate each 2 seconds when it is not found in certificate inventory
+	timeout := time.Duration(60) * time.Second
+
+	startTime := time.Now()
+	for {
+		statusCode, _, body, err := c.request("GET", url, nil)
 		if err != nil {
-			return nil, fmt.Errorf("failed to parse search results: %s, body: %s", err, body)
+			return nil, err
 		}
-		return res, nil
-	default:
-		if body != nil {
-			respErrors, err := parseResponseErrors(body)
-			if err == nil {
-				respError := fmt.Sprintf("unexpected status code on Venafi Cloud certificate search. Status: %d\n", statusCode)
-				for _, e := range respErrors {
-					respError += fmt.Sprintf("Error Code: %d Error: %s\n", e.Code, e.Message)
+
+		switch statusCode {
+		case http.StatusOK:
+			var res = &managedCertificate{}
+			err = json.Unmarshal(body, res)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse search results: %s, body: %s", err, body)
+			}
+			return res, nil
+		default:
+			if body != nil {
+				respErrors, err := parseResponseErrors(body)
+				if err == nil {
+					if statusCode == http.StatusNotFound {
+						if time.Now().After(startTime.Add(timeout)) {
+							return nil, endpoint.ErrRetrieveCertificateTimeout{CertificateID: certificateId}
+						}
+					} else {
+						respError := fmt.Sprintf("unexpected status code on Venafi Cloud certificate search. Status: %d\n", statusCode)
+						for _, e := range respErrors {
+							respError += fmt.Sprintf("Error Code: %d Error: %s\n", e.Code, e.Message)
+						}
+						return nil, errors.New(respError)
+					}
```

Comment:
It's seeming this block of code is doing very similar tasks to the block of code in line 1310 - 1315 with slightly difference in the last part which I consider should be equals in both cases